File: 2021/go/day01/first/first.go
package main

import (
	"bufio"
	"log"
	"os"
	"strconv"
)

func main() {
	counter := 0
	previous := 0

	file, err := os.Open("../input.txt")

	if err != nil {
		log.Fatalf("Can not open file")

	}

	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		log.Println(scanner.Text())
		deep, _ := strconv.Atoi(scanner.Text())
		if previous > 0 && deep > previous {
			counter++
		}
		previous = deep
	}

	log.Println(counter)
}

//Answer: 1681

File: 2015/python/day03/first.py
f = open("input.txt")
dict = {"0,0":1}	
i=0
j=0
while True:
	c = f.read(1)
	if not c:
		break
	if c == '^':
		i+=1
	if c == 'v':
		i-=1
	if c == '>':
		j+=1
	if c == '<':
		j-=1

	key = str(i)+','+str(j)	
	if not key in dict:
		dict[key] = 1
	else:
		dict[key] +=1

print(len(dict))								 	